src/test/fuzz/util.cpp

```diff
@@ -14,6 +14,20 @@
 
 #include <memory>
 
+CPubKey ConsumePubKey(FuzzedDataProvider& fuzzed_data_provider, const bool compressed) noexcept
+{
+    uint8_t pk_type;
+    if (compressed) {
+        pk_type = fuzzed_data_provider.PickValueInArray({0x02, 0x03});
+    } else {
+        pk_type = fuzzed_data_provider.PickValueInArray({0x04, 0x06, 0x07});
+    }
+    std::vector<uint8_t> pk_data = ConsumeFixedLengthByteVector(fuzzed_data_provider, pk_type >= 0x04 ? 65 : 33);
+    pk_data[0] = pk_type;
+    CPubKey pk{pk_data.begin(), pk_data.end()};
```

Comment:
Maybe add a comment to this helper function that the resulting key is not necessarily valid. Or add a `const bool valid` argument which loops until `IsValid()` passes.

Reply:
I don't think we need / want strictly valid pubkeys in a fuzzing context. The point of this function is to return "well-formed" but not necessarily valid pubkeys. By well-formed, I mean they have some sort of parity byte, indicating even or oddness, which also indicates the type of pubkey: compressed, uncompressed, hybrid.